### ISSUE TITLE
There are actually 3 ways to relate temperatures

### ISSUE THREAD

**Issue by hexadecagram** [hgrecco/pint#1567]
These are:

- base unit,
- delta, and
- **magnitude.**

Although Pint is fully capable of performing all 3 calculations, the [documentation](https://pint.readthedocs.io/en/latest/nonmult.html) only mentions the first two. A prime example of when using magnitude (a.k.a. absolute value) is necessary is when approximating the [vapor pressure of water](https://en.wikipedia.org/wiki/Vapour_pressure_of_water).

The way of course to access this is through the `.magnitiude` variable but it seems this happens frequently enough that a (pseudo-?) unit, such as `|degC|`, may be in order (the notation `|...|` being one way of expressing "magnitude"). But in any case it should probably be mentioned in the documenatition regarding temperature.

**Comment by hexadecagram** [comment]
As an example, here is a snippet of code that computes the Arden Buck equation:

```
    def vapor_pressure(self, T):
        Tprime = T.to(self.ureg.degC)
        self.__log.debug('Tprime=%s', T.to(self.ureg.degC))
        p_s = self.Q_(257.14, self.ureg.delta_degC) + Tprime
        self.__log.debug('p_s=%s', p_s)
        p_s = Tprime.magnitude / p_s.magnitude  # A third way!
        self.__log.debug('p_s=%s', p_s)
        p_s_1 = 18.678 - Tprime.magnitude / 234.5
        self.__log.debug('p_s_1=%s', p_s_1)
        p_s *= p_s_1
        self.__log.debug('p_s=%s', p_s)
        p_s = self.Q_(6.1121, self.ureg.hPa) * exp(p_s)
        return p_s
```

**Comment by hexadecagram** [comment]
As an example, here is a snippet of code that computes the Arden Buck equation:

```
    def vapor_pressure(self, T):
        Tprime = T.to(self.ureg.degC)
        p_s = self.Q_(257.14, self.ureg.delta_degC) + Tprime
        p_s = Tprime.magnitude / p_s.magnitude  # A third way!
        p_s_1 = 18.678 - Tprime.magnitude / 234.5
        p_s *= p_s_1
        p_s = self.Q_(6.1121, self.ureg.hPa) * exp(p_s)
        return p_s
```

**Comment by hgrecco** [comment]
I do not fully understand the proposal. You are suggesting to incorporate a new syntax? How would it look?

(Remember that `.magnitude` can also be written as `.m`)

**Comment by hexadecagram** [comment]
I should have been more explicit. I am requesting that the [documentation that I linked to](https://pint.readthedocs.io/en/latest/nonmult.html) be updated to mention that there are actually 3 ways to perform arithmetic on teperature quantities.

As I was working through that (and similar formulas), I spent a lot of time reading over that documentation and trying to make degC and delta_degC work, when it turned out that a call to `magnitude()` (or `m()`) was the right thing to do.

I wouldn't mind forking, making the edits myself, and offering my commits, but I thought I should discuss it with you first.

**Comment by dalito** [comment]
If you are using magnitude you are leaving the safety net that pint provides. It is a way to bypass unit checks/conversions. I would not call it "a third way to relate temperatures". magnitude is not special to temperatures.

**Comment by hgrecco** [comment]
I agree with @dalito. I am closing this now, feel free to reopen should you have any question.

**Comment by hexadecagram** [comment]
If that's true, then why is it that some equations in Physics (such as the one I mentioned) requires the magnitude of temperature?

I agree with your perspective where the scope of Pint is concerned, but shouldn't something like this at least warrant a mention?

**Comment by hexadecagram** [comment]
I would urge you to consider this one carefully. Temperature in particular is a tricky thing to work with in my experience. Vapor pressure in particular requires approximations such as Arden-Buck because the science just "isn't there" yet. Have a close look at that Wikipedia page and consider how each of those equations are making use of the magnitude of temperature.

In any case, I've definitely learned a lot by having to work through this. I just figured some users might appreciate being able to avoid the struggles I had.

If it still is beyond the scope of Pint, so be it. Thanks for your consideration.

**Comment by hgrecco** [comment]
What is not clear to me is why you are singling out temperature. Numerical-value equations (like the Arden-Buck equation) are present in all disciplines, including different dimensionality. 

And because they are numerical-value equations, they say that each component should be expressed in certain units that is why you have to use the magnitude. It is not a Pint/Python thing. 

By the way, I would write your code in the following manner:

```python
def arden_buck(temperature):
    temp = temperature.m_as(ureg.degC)
    value = (18.678 - temp/234.5) * temp / (257.14 + temp)
    result = 6.1121 * exp(value)
    return result * ureg.hPa
```

that clearly shows that this is a numerical-value equation as in the two inner most lines only numbers appear (not quantities). 

Or use wraps which is exactly why it was made for:
```python
@ureg.wraps(ureg.hPa, ureg.degC)
def arden_buck(temperature):
    value = (18.678 - temperature/234.5) * temperature / (257.14 + temperature)
    return 6.1121 * exp(value)
```

But again, it is not clear to me is why you are singling out temperature. This appears in many other cases and should be only used when needed. Quantity equations are always better than numerical-value equations.

**Comment by dopplershift** [comment]
That equation can also be made dimensionally sound by giving all of the constants their correct dimensionality. Given that the equation works for degC, that means there is some implied dimensionality about each of those constants such that `f(value): degC -> dimensionless.`

**Comment by hexadecagram** [comment]
> What is not clear to me is why you are singling out temperature. Numerical-value equations (like the Arden-Buck equation) are present in all disciplines, including different dimensionality.

I wouldn't say I am **intentionally** singling out temperature, but whenever I have worked through equations in Physics or Chemistry, whenever something like the norm of a vector is called for, it is usually quite explicit right in the equation. When I worked through Arden-Buck, it was **not** explicit. So I would say it's more like I'm singling out Arden-Buck.

I will admit that I am very much an amateur scientist, so I could be completely overlooking something about those equations that would have made it clear to effectively just "drop" the units off of the quantity and plug the numbers straight in. In any case, they all seem like back-of-the envelope calculations that really are just approximations with little attention to dimensions. I'm also fairly new to Pint, to be honest. ;-)

@dopplershift, I spent nearly a full day trying to do just that and could not get it to balance out. I posted exactly what I managed to boil it down to in the end. It seems @hgrecco managed to succeed where I failed.

Thanks for hearing me out but at this point even I am agreeing that my suggestion really is outside the scope of Pint.
